xbmc/platform/win32/filesystem/Win32File.cpp

```diff
@@ -577,8 +576,9 @@ int CWin32File::Stat(const CURL& url, struct __stat64* statData)
     if (lastDot != std::wstring::npos && pathnameW.rfind(L'\\') < lastDot)
     { // file has some extension
       const std::wstring fileExt(pathnameW, lastDot);
-      XUTILS::auto_buffer buf(32767 * sizeof(wchar_t)); // maximum possible size
-      const DWORD envRes = GetEnvironmentVariableW(L"PATHEXT", (wchar_t*)buf.get(), buf.size() / sizeof(wchar_t));
+      std::vector<char> buf(32767 * sizeof(wchar_t)); // maximum possible size
```

Comment:
Change to `std::vector<wchar_t>` and get rid of the casts and the extra size calculation?

Reply:
done